src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.ExecutionStatus
+import cromwell.engine.workflow.WorkflowActor.ExecutionStore
+
+sealed trait ExecutionStoreKey {
+  val scope: Scope
+
+  val index: Option[Int]
+
+  val parent: Option[ExecutionStoreKey]
+}
+
+
+case class CallKey(override val scope: Call,
```

Comment:
Some java folks prefer the concise

```java
public static void main(String[] args)
```

Others insist it _must_ be

```java
public static void main(final String[] args)
```

or your PR will be rejected forthwith, as the latter is "correct". The same with `@Override` on java interface implementations, etc.

I'm hearing that RedTeam doesn't like this particular type of extra compiler checks, and code hints, so I'm fine taking them out.